pkgs/tools/inputmethods/fcitx5/with-addons.nix

```
@@ -36,13 +36,14 @@ symlinkJoin {
       wrapProgram $out/bin/fcitx5-config-qt --prefix FCITX_ADDON_DIRS : "$out/lib/fcitx5"
     ''}
 
-    desktop=share/applications/org.fcitx.Fcitx5.desktop
-    autostart=etc/xdg/autostart/org.fcitx.Fcitx5.desktop
-    rm $out/$desktop
-    rm $out/$autostart
-    cp ${fcitx5}/$desktop $out/$desktop
-    sed -i $out/$desktop -e "s|^Exec=.*|Exec=$out/bin/fcitx5|g"
-    ln -s $out/$desktop $out/$autostart
+    pushd $out
+    grep_dirs=(share/applications etc/xdg/autostart share/dbus-1/services)
+    grep -Rl --include=\*.{desktop,service} "''${grep_dirs[@]}" -e ${fcitx5} | while read -r file; do
```

```suggestion
    grep -Rl --include=\*.{desktop,service} share/applications etc/xdg/autostart share/dbus-1/services -e ${fcitx5} | while read -r file; do
```
we can really easily avoid quoting hell here

---

done